src/System.Net.Primitives/src/System/Net/IPEndPoint.cs

```diff
@@ -148,11 +148,13 @@ public override EndPoint Create(SocketAddress socketAddress)
 
         public override bool Equals(object comparand)
         {
-            if (!(comparand is IPEndPoint))
+            var other = comparand as IPEndPoint;
```

Comment:
`if (!(comparand is IPEndPoint other))` ?

Reply:
Similar to @JonHanna 

```c#
bool isEqual = false;
if (comparand is IPEndPointA other)
{
    isEqual = other._address.Equals(_address) && other._port == _port;
}

return isEqual;
```

Reply:
Way to verbose. It's just

```C#
return comparand is IPEndPoint other && other._address.Equals(_address) && other._port == _port;
```

Reply:
@mikedn  for coding-guidelines: would you implement this as expression bodied method or with the explicit return?

Reply:
> for coding-guidelines: would you implement this as expression bodied method or with the explicit return?

I think it's a matter of taste. I'd say no because the expression is rather long and may need to be moved on a separate line thus diminishing the advantage of using expression bodied methods. Others may have a different opinion.